sdk/src/transactions/scheduler.rs

```diff
@@ -0,0 +1,1243 @@
+        workflow::Scheduler::CHECK_TIME_CONSTRAINT.name.into(),
+        vec![],
+        vec![task, clock],
+    ))
+}
+
+/// PTB template to register DAG execution config on the execution policy.
+pub fn register_begin_execution(
+    tx: &mut sui::ProgrammableTransactionBuilder,
+    objects: &NexusObjects,
+    policy: sui::Argument,
+    config: sui::Argument,
+) -> anyhow::Result<sui::Argument> {
+    Ok(tx.programmable_move_call(
+        objects.workflow_pkg_id,
+        workflow::DefaultTap::REGISTER_BEGIN_EXECUTION.module.into(),
+        workflow::DefaultTap::REGISTER_BEGIN_EXECUTION.name.into(),
+        vec![],
+        vec![policy, config],
+    ))
+}
+
+/// PTB template to invoke DAG execution from the scheduler via the Default TAP.
+pub fn dag_begin_execution_from_scheduler(
+    tx: &mut sui::ProgrammableTransactionBuilder,
+    objects: &NexusObjects,
+    task: &sui::ObjectRef,
+    dag: &sui::ObjectRef,
+    leader_cap: sui::Argument,
+    claim_coin: sui::Argument,
+    amount_execution: u64,
+    amount_priority: u64,
+) -> anyhow::Result<sui::Argument> {
+    // `self: &mut DefaultTAP`
+    let tap = tx.obj(sui::ObjectArg::SharedObject {
+        id: objects.default_tap.object_id,
+        initial_shared_version: objects.default_tap.version,
+        mutable: true,
+    })?;
+
+    // `task: &mut Task`
+    let task = shared_task_arg(tx, task)?;
+
+    // `dag: &DAG`
+    let dag = tx.obj(sui::ObjectArg::SharedObject {
+        id: dag.object_id,
+        initial_shared_version: dag.version,
+        mutable: false,
+    })?;
+
+    // `gas_service: &mut GasService`
+    let gas_service = tx.obj(sui::ObjectArg::SharedObject {
+        id: objects.gas_service.object_id,
+        initial_shared_version: objects.gas_service.version,
+        mutable: true,
+    })?;
+
+    // `amount_execution: u64`
+    let amount_execution_arg = tx.pure(amount_execution)?;
+
+    // `amount_priority: u64`
+    let amount_priority_arg = tx.pure(amount_priority)?;
+
+    // `clock: &Clock`
+    let clock = tx.obj(sui::CLOCK_OBJ_ARG)?;
+
+    Ok(tx.programmable_move_call(
+        objects.workflow_pkg_id,
+        workflow::DefaultTap::DAG_BEGIN_EXECUTION_FROM_SCHEDULER
+            .module
+            .into(),
+        workflow::DefaultTap::DAG_BEGIN_EXECUTION_FROM_SCHEDULER
+            .name
+            .into(),
+        vec![],
+        vec![
+            tap,
+            task,
+            dag,
+            gas_service,
+            leader_cap,
+            claim_coin,
+            amount_execution_arg,
+            amount_priority_arg,
+            clock,
+        ],
+    ))
+}
+
+/// PTB helper that consumes the next scheduled occurrence and invokes the TAP.
+pub fn execute_scheduled_occurrence(
+    tx: &mut sui::ProgrammableTransactionBuilder,
+    objects: &NexusObjects,
+    task: &sui::ObjectRef,
+    dag: &sui::ObjectRef,
+    leader_cap: sui::Argument,
+    claim_coin: sui::Argument,
```

Comment:
these can easily be `&sui::ObjectRef` as well, right? I think it should be this function's responsibility to turn them into `Argument`. Same for the other functions here.